test/Garnet.test/RespSortedSetTests.cs

```diff
@@ -341,6 +341,29 @@ public void AddScore()
             Assert.AreEqual(expectedResponse, actualValue);
         }
 
+        [Test]
+        public void CanGetMemberScores()
```

Comment:
Please check the following scenario as I see an issue here:
(note that key2 does not exist in the set)

127.0.0.1:3278> zmscore key2 "one"
(nil)
127.0.0.1:3278> zmscore key2 "one"
(error) ERR unknown command

Reply:
This was a tricky one due to my lack of familiarity with the project. I was not "draining" the input in the case of the key not being found. This was resulting in the `ptr` not advancing and thus the `readHead` not advancing correctly..

I have a couple thoughts on this:

**Testing calling twice now seems moot**

I've resolved it here but are there tests patterns in place that would catch this issue on any command, or more generally?

```
127.0.0.1:3278> ZMSCORE key2 one
(error) WRONGTYPE Operation against a key holding the wrong kind of value.
127.0.0.1:3278> ZMSCORE key2 one
(error) WRONGTYPE Operation against a key holding the wrong kind of value.
127.0.0.1:3278> ZMSCORE key2 one
(error) WRONGTYPE Operation against a key holding the wrong kind of value.
```

**The existing patterns for constructing obj methods seems error prone**

Each method needing to individually implement the logic to advance the `readHead` properly is very tricky. Particularly as there are variations between methods on how it gets advanced, how `ptr` gets advanced, and etc.

I'm wondering in what circumstances it shouldn't be advanced to the end of the input command after object commands complete? Or rather, if there is some way to handle this reliably in one place.

Reply:
Good catch! 

**Testing calling twice now seems moot** - you're correct, let me bring this up with the team and see what they think...

**The existing patterns for constructing obj methods seems error prone** - you're VERY correct, in fact, one of our engineers is working on an overhaul for this part of the code, so should get better pretty soon! :)